apps/web/src/utils/codeSnippets.ts

```diff
@@ -1,16 +1,23 @@
 import { API_ROOT } from '../config/index';
 
-export const createNodeSnippet = (
-  identifier: string,
-  to: Record<string, unknown>,
-  payload: Record<string, unknown>,
-  overrides?: Record<string, unknown>,
-  snippet?: Record<string, unknown>,
-  apiKey = '<API_KEY>'
-) => {
+export type CodeSnippetProps = {
+  identifier: string;
+  to: Record<string, unknown>;
+  payload: Record<string, unknown>;
+  overrides?: Record<string, unknown>;
+  snippet?: Record<string, unknown>;
+  secretKey?: string;
+  bridgeUrl?: string;
+};
+
+const SECRET_KEY_ENV_KEY = 'NOVU_SECRET_KEY';
+
+export const createNodeSnippet = ({ identifier, to, payload, overrides, snippet, secretKey }: CodeSnippetProps) => {
+  const renderedSecretKey = secretKey ? `'${secretKey}'` : `process.env['${SECRET_KEY_ENV_KEY}']`;
```

Comment:
A pattern that renders the secret key if present (i.e. in Local mode), and falls back to the environment access API for the language, so that Production keys stay hidden.